test/end2end_test.go

```diff
@@ -7668,3 +7668,82 @@ func (s) TestClientSettingsFloodCloseConn(t *testing.T) {
 	s.GracefulStop()
 	timer.Stop()
 }
+
+// TestDeadlineSetOnConnectionOnClientCredentialHandshake tests that there is a deadline
+// set on the net.Conn when a credential handshake happens in http2_client.
+func (s) TestDeadlineSetOnConnectionOnClientCredentialHandshake(t *testing.T) {
+	lis, err := net.Listen("tcp", "localhost:50053")
+	if err != nil {
+		t.Fatalf("Failed to listen: %v", err)
+	}
+	go func() {
+		lis.Accept()
+	}()
```

Comment:
This is equivalent to `go lis.Accept()`

Reply:
Noted. Changed.

Reply:
......aaaaaaaaand now that I've told you this and thought about it a little more, I think it's probably also important to observe the `Conn` you get back and `Close` it.  Sorry!

Reply:
Why? This test is validating that there's a deadline set on the conn on the client side credential handshake. What does closing the conn do?

Reply:
It's a leaked resource if you don't close it.

Reply:
Added.